phenopacketlab-restapi/src/test/java/org/monarchinitiative/phenopacketlab/restapi/controller/ValidateControllerTest.java

```diff
@@ -0,0 +1,295 @@
+                "    }\n" +
+                "  }],\n" +
+                "  \"diseases\": [{\n" +
+                "    \"term\": {\n" +
+                "      \"id\": \"NCIT:C4024\",\n" +
+                "      \"label\": \"Esophageal Squamous Cell Carcinoma\"\n" +
+                "    },\n" +
+                "    \"clinicalTnmFinding\": [{\n" +
+                "      \"id\": \"NCIT:C48724\",\n" +
+                "      \"label\": \"T2 Stage Finding\"\n" +
+                "    }, {\n" +
+                "      \"id\": \"NCIT:C48706\",\n" +
+                "      \"label\": \"N1 Stage Finding\"\n" +
+                "    }, {\n" +
+                "      \"id\": \"NCIT:C48699\",\n" +
+                "      \"label\": \"M0 Stage Finding\"\n" +
+                "    }]\n" +
+                "  }],\n" +
+                "  \"metaData\": {\n" +
+                "    \"created\": \"2021-05-14T10:35:00Z\",\n" +
+                "    \"createdBy\": \"anonymous biocurator\",\n" +
+                "    \"resources\": [{\n" +
+                "      \"id\": \"ncit\",\n" +
+                "      \"name\": \"NCI Thesaurus\",\n" +
+                "      \"url\": \"http://purl.obolibrary.org/obo/ncit.owl\",\n" +
+                "      \"version\": \"21.05d\",\n" +
+                "      \"namespacePrefix\": \"NCIT\",\n" +
+                "      \"iriPrefix\": \"http://purl.obolibrary.org/obo/NCIT_\"\n" +
+                "    }, {\n" +
+                "      \"id\": \"efo\",\n" +
+                "      \"name\": \"Experimental Factor Ontology\",\n" +
+                "      \"url\": \"http://www.ebi.ac.uk/efo/efo.owl\",\n" +
+                "      \"version\": \"3.34.0\",\n" +
+                "      \"namespacePrefix\": \"EFO\",\n" +
+                "      \"iriPrefix\": \"http://purl.obolibrary.org/obo/EFO_\"\n" +
+                "    }, {\n" +
+                "      \"id\": \"uberon\",\n" +
+                "      \"name\": \"Uber-anatomy ontology\",\n" +
+                "      \"url\": \"http://purl.obolibrary.org/obo/uberon.owl\",\n" +
+                "      \"version\": \"2021-07-27\",\n" +
+                "      \"namespacePrefix\": \"UBERON\",\n" +
+                "      \"iriPrefix\": \"http://purl.obolibrary.org/obo/UBERON_\"\n" +
+                "    }, {\n" +
+                "      \"id\": \"ncbitaxon\",\n" +
+                "      \"name\": \"NCBI organismal classification\",\n" +
+                "      \"url\": \"http://purl.obolibrary.org/obo/ncbitaxon.owl\",\n" +
+                "      \"version\": \"2021-06-10\",\n" +
+                "      \"namespacePrefix\": \"NCBITaxon\",\n" +
+                "      \"iriPrefix\": \"http://purl.obolibrary.org/obo/NCBITaxon_\"\n" +
+                "    }],\n" +
+                "    \"phenopacketSchemaVersion\": \"2.0\"\n" +
+                "  }\n" +
+                "}";
+
+    private final String INCORRECT_PHENOPACKET =
+                "{\n" +
```

Comment:
There could be a few more band cases in the tests. Are there any ways you can think of which could break the code with bad packets? Can tests pin these down?

Reply:
yeah there are a lot of possibilities as the schema is pretty large. I'll add a couple more smaller phenopackets testing different edge cases, but here Im not testing the validation (this is the job of phenopacket-tools on which I depend), Im just testing that the endpoint when it returns a non-validation.